From 93befdfdbf29b17b6225ef90b84390a8a1b3af8f Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Sat, 28 Jul 2012 06:08:24 +0800
Subject: [PORT FROM MAIN]RTSPSouce: add check for mHandler in release

BZ: 49222

fix the media server issue in RTSP release
duplicate quit msg may cause mHandler null and it may cause crash in onDisconnect

orig-patch:
55909(BZ:44103),RTSPSouce: add check for mHandler in release

Change-Id: Ie00a13a5209375177ba6918f66436adf0f55e575
Orig-Change-Id: I338c13973a536bbcf7666ed4544a276d44057555
Signed-off-by: ywan171 <yi.a.wang@intel.com>
---
 .../libmediaplayerservice/nuplayer/RTSPSource.cpp  |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/media/libmediaplayerservice/nuplayer/RTSPSource.cpp b/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
index a5ff0ca..d887829 100644
--- a/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
+++ b/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
@@ -636,8 +636,10 @@ void NuPlayer::RTSPSource::onDisconnected(const sp<AMessage> &msg) {
     CHECK(msg->findInt32("result", &err));
     CHECK_NE(err, (status_t)OK);
 
-    mLooper->unregisterHandler(mHandler->id());
-    mHandler.clear();
+    if (mHandler != NULL) {
+        mLooper->unregisterHandler(mHandler->id());
+        mHandler.clear();
+    }
 
     if (mState == CONNECTING) {
         // We're still in the preparation phase, signal that it
-- 
1.7.4.1

